src/Raven.Server/Documents/PeriodicBackup/BackupUploader.cs

```diff
@@ -0,0 +1,328 @@
+                    NativeMemory.EnsureRegistered();
+
+                    using (localUploadStatus.UpdateStats(_isFullBackup))
+                    using (var fileStream = File.OpenRead(_backupUploaderSettings.BackupPath))
+                    {
+                        var uploadProgress = localUploadStatus.UploadProgress;
+                        try
+                        {
+                            localUploadStatus.Skipped = false;
+                            uploadProgress.ChangeState(UploadState.PendingUpload);
+                            uploadProgress.SetTotal(fileStream.Length);
+
+                            AddInfo($"Starting {uploadStatus.GetType().AssemblyQualifiedName}");
+
+                            var bytesPutsPerSec = new MeterMetric();
+
+                            long lastUploadedInBytes = 0;
+                            var totalToUpload = new Size(uploadProgress.TotalInBytes, SizeUnit.Bytes).ToString();
+                            var sw = Stopwatch.StartNew();
+                            var progress = new Progress(uploadProgress)
+                            {
+                                OnUploadProgress = () =>
+                                {
+                                    if (sw.ElapsedMilliseconds <= 1000)
+                                        return;
+
+                                    var totalUploadedInBytes = uploadProgress.UploadedInBytes;
+                                    bytesPutsPerSec.MarkSingleThreaded(totalUploadedInBytes - lastUploadedInBytes);
+                                    lastUploadedInBytes = totalUploadedInBytes;
+                                    var uploaded = new Size(totalUploadedInBytes, SizeUnit.Bytes);
+                                    uploadProgress.BytesPutsPerSec = bytesPutsPerSec.MeanRate;
+                                    AddInfo($"Uploaded: {uploaded} / {totalToUpload}");
+                                    sw.Restart();
+                                }
+                            };
+
+                            uploadToServer(settings, fileStream, progress);
+
+                            AddInfo($"Total uploaded: {totalToUpload}, took: {MsToHumanReadableString(uploadProgress.UploadTimeInMs)}");
+                        }
+                        finally
+                        {
+                            uploadProgress.ChangeState(UploadState.Done);
+                        }
+                    }
+                }
+                catch (OperationCanceledException e)
+                {
+                    // shutting down
+                    localUploadStatus.Exception = e.ToString();
+                    _exceptions.Add(e);
+                }
+                catch (Exception e)
+                {
+                    localUploadStatus.Exception = e.ToString();
+                    _exceptions.Add(new InvalidOperationException($"Failed to backup to {uploadStatus.GetType().FullName}", e));
```

Comment:
You can replace uploadStatus.GetType().FullName with targetName

Reply:
### done